x-pack/metricbeat/module/awsfargate/task_stats/task_stats_integration_test.go

```diff
@@ -65,7 +66,14 @@ func TestData(t *testing.T) {
 	formattedStats := getStatsList(taskStatsOutput, taskOutput)
 	assert.Equal(t, 1, len(formattedStats))
 	event := createEvent(&formattedStats[0])
-	standardizeEvent := m.StandardizeEvent(event)
 
+	// Build a metricset to test the event
+	metricSet := mbtest.NewReportingMetricSetV2Error(t, config)
+
+	// The goal here is to make sure every element inside the
+	// event has a matching field ("no field left behind").
+	mbtest.TestMetricsetFieldsDocumented(t, metricSet, []mb.Event{event})
+
+	standardizeEvent := m.StandardizeEvent(event)
```

Comment:
@kaiyan-sheng I started adding the check for the undocumented fields here (we already loaded the event from the JSON files). 

It works, but it's probably worth adding a proper `TestFetch()` for this kind of check, as you originally suggested, isn't it? What do you think?

Reply:
Not sure if it has to be this way, but usually we have TestFetch function for TestMetricsetFieldsDocumented and TestData to WriteEvents.

Reply:
You're right: I am moving the `TestMetricsetFieldsDocumented()` to the `TestFetch` function; this case isn't special enough to make an exception.

Reply:
Moved!